### ISSUE TITLE
`transmutes_expressible_as_ptr_casts` does not detect reference-to-usize transmutes

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust-clippy#8622]
### Summary

Clippy by default lints against transmuting raw pointers to `usize`. However, it does not lint against transmuting *references* to `usize`. Both of these transmutes are [highly questionable](https://github.com/rust-lang/unsafe-code-guidelines/issues/286#issuecomment-1085144431) so it'd be good to lint them all.

### Lint Name

transmutes_expressible_as_ptr_casts

### Reproducer

I tried this code:

```rust
fn main() { unsafe {
    let _x: usize = std::mem::transmute(&0);
}}
```

I expected to see this happen: clippy should complain about the transmute.

Instead, this happened: clippy remains silent.

### Version

```text
Current playground
```
